### PR TITLE
Add db experimental VM

### PR DESCRIPTION
**Pull request by r-ash**
This is for testing out db improvements for VIMC 2.0

### CODE REVIEW

File: db-experiment/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Vagrant.configure(2) do |config|
+  config.vm.box = box
+  config.vm.network "public_network", bridge: "External Switch", auto_config: true
+  config.vm.provider :hyperv do |hyperv|
+    hyperv.mac = mac
+    hyperv.vmname = hostname
+    hyperv.cpus = cpus
+    hyperv.maxmemory = ram
+    hyperv.memory = "65536"
```
Comment (richfitz) [inline db-experiment/Vagrantfile]: Looks like this could go to 256GB I think?

File: db-experiment/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# -*- mode: ruby -*-
+
+box = "generic/ubuntu1804"
+ram = "65536"
+hostname = "wpia-db-experiment"
+mac = "00:15:5d:1a:84:10"
+cpus = 2
```
Comment (richfitz) [inline db-experiment/Vagrantfile:7]: Looks like're already 8 cpus over budget over the machine! So I guess 2 is the min here, but something that we might look at soon

File: db-experiment/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# -*- mode: ruby -*-
+
+box = "generic/ubuntu1804"
+ram = "65536"
```
Comment (richfitz) [inline db-experiment/Vagrantfile]: unused constant

**Review by richfitz — CHANGES_REQUESTED** [review]

File: db-experiment/Vagrantfile (hunk 1 above)
Comment (r-ash) [reply db-experiment/Vagrantfile]: Upped that to 256

File: db-experiment/Vagrantfile (hunk 2 above)
Comment (r-ash) [reply db-experiment/Vagrantfile:7]: What does this core count add up to?

We have 2 24 core processors multithreaded to 48 each. So are we already using 34/96?

File: db-experiment/Vagrantfile (hunk 3 above)
Comment (r-ash) [reply db-experiment/Vagrantfile]: Use for the `hyperv.maxmemory` but value same as `hyperv.memory`. Copied this from another VM setup, I'm not sure how they are different. Wes do you know?

File: db-experiment/Vagrantfile (hunk 2 above)
Comment (richfitz) [reply db-experiment/Vagrantfile:7]: 2 12 core processors, 24 real cores, 48 hyperthreaded I think (based on a google)

Comment (weshinsley) [reply db-experiment/Vagrantfile:7]: Two of these beasts - https://ark.intel.com/content/www/us/en/ark/products/192481/intel-xeon-platinum-8268-processor-35-75m-cache-2-90-ghz.html

48 real cores, 96 hyperthreaded

File: db-experiment/Vagrantfile (hunk 3 above)
Comment (weshinsley) [reply db-experiment/Vagrantfile]: hyperv.memory is allocated at startup, whereas hyperv.maxmemory is the max limit. If both are set, I'd set them to be the same - I'm not convinced dynamic memory allocation will help us on this host...

...but in that case, possibly, we should just not set maxmemory at all, which makes hyperv.memory work as a static allocation anyway.

Comment (weshinsley) [reply db-experiment/Vagrantfile]: So I'm not 100% sure why both are set - there might have been a bug in the config handling of earlier vagrant, which forced me to set them both even if I only wanted to set static.

File: db-experiment/Vagrantfile (hunk 2 above)
Comment (weshinsley) [reply db-experiment/Vagrantfile:7]: In any case, with our server-type VMs, total CPU usage of all the VMs currently running is 1% (and the windows scheduler isn't allocating particular cores to particular VMs I think?) . I wonder what it looks like when the VMs are more stressed?

**Review by weshinsley — APPROVED** [review]

**Review by richfitz — APPROVED** [review]

**Pull request merged**